src/Tizen.NUI/src/internal/Xaml/CreateValuesVisitor.cs

```diff
@@ -449,20 +449,50 @@ static object CreateLanguagePrimitive(Type nodeType, IElementNode node)
                 if (nodeType == typeof(Int16))
                 {
                     short retval;
-                    if (short.TryParse(valuestring, NumberStyles.Number, CultureInfo.InvariantCulture, out retval))
+                    if (valuestring.EndsWith("dp"))
+                    {
+                        return (short)GraphicsTypeManager.Instance.ConvertToPixel(short.Parse(valuestring.Substring(0, valuestring.LastIndexOf("dp")), CultureInfo.InvariantCulture));
+                    }
+                    else if (valuestring.EndsWith("px"))
+                    {
+                        return (short.Parse(valuestring.Substring(0, valuestring.LastIndexOf("px")), CultureInfo.InvariantCulture));
+                    }
+                    else if (short.TryParse(valuestring, NumberStyles.Number, CultureInfo.InvariantCulture, out retval))
+                    {
                         return retval;
+                    }
```

Comment:
메소드로 추출해서 리펙토링을 하면 좋을것 같습니다.

그리고 double을 리턴하는 메소드 하나만 뽑아서 나머지는 캐스팅해서 사용하면 될것 같습니다.
```suggestion
                    retval = (short) GraphicsTypeManager.Instance.ParseScableUnit(valuestring);
```

``` c#
        public static double ParseScableUnit(string valuestring)
        {
            if (valuestring == null)
            {
                return 0;
            }
            else if (valuestring.EndsWith("dp"))
            {
                var retval = double.Parse(valuestring.Substring(0, valuestring.LastIndexOf("dp")), CultureInfo.InvariantCulture);
                return retval * GraphicsTypeManager.Instance.Density;
            }
            else if (valuestring.EndsWith("px"))
            {
                return double.Parse(valuestring.Substring(0, valuestring.LastIndexOf("px")), CultureInfo.InvariantCulture);
            }
            else if (double.TryParse(valuestring, NumberStyles.Number, CultureInfo.InvariantCulture, out double retval))
            {
                return retval;
            }
            return 0;
        }
```